engine/api/application_deployment.go

```diff
@@ -186,9 +195,14 @@ func (api *API) getApplicationDeploymentStrategyConfigHandler() service.Handler
 			opts = []application.LoadOptionFunc{application.LoadOptions.WithClearDeploymentStrategies}
 		}
 
-		app, err := application.LoadByName(api.mustDB(), key, appName, opts...)
+		proj, err := project.Load(api.mustDB(), projectKey)
```

Comment:
I think it's a bit sad to load the project to get its ID, whereas the projectKey is an identifier.
we should consider application.LoadByProjectKeyAndName